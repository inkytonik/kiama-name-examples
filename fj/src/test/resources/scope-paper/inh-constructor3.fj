// ADAPTED FROM inheritance/constructors.spt:inherit unary constructor class
// FIX: Add fields to A
class A extends Object {
  Object f;
  A(Object f) {
    super();
    this.f = f;
  }
}
class B extends A {
  B(Object f) {
    super(f);
  }
}
