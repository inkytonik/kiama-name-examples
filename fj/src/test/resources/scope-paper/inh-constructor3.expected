src/test/resources/scope-paper/inh-constructor3.fj:4:10:info: f def: Object
  Object f;
         ^
src/test/resources/scope-paper/inh-constructor3.fj:5:12:info: f def: Object
  A(Object f) {
           ^
src/test/resources/scope-paper/inh-constructor3.fj:11:12:info: f def: Object
  B(Object f) {
           ^
